### ISSUE TITLE
Static middleware tries to serve directory and server dies

### ISSUE THREAD

**Issue by aj0strow** [divshot/superstatic#155]
Ran into an interesting issue. Ideally clean routes shouldn't need to worry about conflicting with folder structure. For example:

```
+ dist
  + dashboard
    + home
  - index.html
```

Static route like:

```json
{ "/**!(.*)": "index.html" }
```

Angular route like:

```javascript
$routeProvider.when('/dashboard', {
  templateUrl: '/dashboard/home/home.html'
})
```

I want `GET /dashboard` to serve `index.html` and for angular to take it from there. Unfortunately the static server tries to serve the `dashboard` directory, and it blows up. 

```
Error: EISDIR, read
```

**Comment by scottcorgan** [comment]
What version are you running?

**Comment by scottcorgan** [comment]
The reason I ask is because I am unable to replicate with that code. There was a bug like this that got fixed in version `2.1.2` in commit https://github.com/divshot/superstatic/commit/ea00359f24946c9757eb48a029c7f351d1008a83.

**Comment by aj0strow** [comment]
My bad, it's fixed in latest. I reverted to avoid the etag situation.
